manifests/reload_service.pp

```diff
@@ -7,10 +7,14 @@
 #
 class consul::reload_service {
 
-  exec { 'reload consul service':
-    path        => [$consul::bin_dir,'/bin','/usr/bin'],
-    command     => 'consul reload',
-    refreshonly => true,
+  # Don't attempt to reload if we're not supposed to be running.
+  # This can happen during pre-provisioning of a node.
+  if $::consul::service_ensure != 'stopped' {
```

Comment:
I think we should make this a little more robust:

```
if $consul::manage_service == true and $consul::service_enable == true and $consul::service_ensure == 'running' {
 exec {...}
}
```

Also no need to prefix the variable with `::`, it can just be `$consul::some_param`